panther_analysis_tool/main.py

```diff
@@ -259,6 +259,83 @@ def upload_analysis(args: argparse.Namespace) -> Tuple[int, str]:
     return 0, ''
 
 
+def update_schemas(args: argparse.Namespace) -> Tuple[int, str]:
+    """Updates managed schemas in a Panther deployment.
+
+    Returns 1 if the update fails.
+
+    Args:
+        args: The populated Argparse namespace with parsed command-line arguments.
+
+    Returns:
+        A tuple of return code and the archive filename.
+    """
+
+    # optionally set env variable for profile passed as argument
+    # this must be called prior to setting up the client
+    if args.aws_profile is not None:
+        logging.info('Using AWS profile: %s', args.aws_profile)
+        set_env("AWS_PROFILE", args.aws_profile)
+
+    client = boto3.client('lambda')
+    logging.info('Fetching updates')
+    response = client.invoke(FunctionName='panther-logtypes-api',
+                             InvocationType='RequestResponse',
+                             Payload=json.dumps(
+                                 {'ListManagedSchemaUpdates': {}}))
+    response_str = response['Payload'].read().decode('utf-8')
+    response_payload = json.loads(response_str)
+
+    api_err = response_payload.get('error')
+    if api_err is not None:
+        logging.error(
+            'Failed to list managed schema updates\n\tcode: %s\n\terror message: %s',
+            api_err['code'], api_err['message'])
+        return 1, ''
+
+    releases = response_payload.get('releases')
+    if releases is None or len(releases) is 0:
```

Comment:
Equivalent: `if not releases`

Note that `is` checks for object identity not equality. The check here happens to be correct, because up to a certain number integer instances are cached. For large numbers this isn't true:

```python
>>> id(0)
4454249536
>>> id(0)
4454249536
>>> id(0)
4454249536
>>> id(93939390303030303030303003030303030303003)
4456905776
>>> id(93939390303030303030303003030303030303003)
4456905728
>>> id(93939390303030303030303003030303030303003)
4456905776
```

Reply:
The way I understand this would be that I check that `relates` is not `null` in the JSON and that it is not `[]` either.
Is this the same as `if not releases:`? How weird that there's a boolean value for lists. Forgot _that_ bug generator existed in Python.

Reply:
fixed

Reply:
As far as I remember all empty objects (sets, tuples, dicts, lists, strings) have a truth value of `False`.

Reply:
@georgepsarakis hit the nail on the head

Python will evaluate `if __OBJECT__` as bool `False` if `__OBJECT__` is the default value or None